test(e2e,playwright): desktop layout

shabeebk opened the pull request:
Checklist:

<!-- Please follow this checklist and put an x in each of the boxes, like this: [x]. It will ensure that our team takes your pull request seriously. -->

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

<!--If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.-->

Ref #51705

<!-- Feel free to add any additional description of changes below this line -->

shabeebk committed:
Add e2e test for desktop-layout component

With this commit, the playwright will test the desktop-layout
component for possible layout with 2 and 3 panes each.

shabeebk committed:
Merge branch 'freeCodeCamp:main' into e2e/classic-challenge-desktop-layout

shabeebk commented:
camperbot cant code

raditotev commented on e2e/desktop-layout-component.spec.ts:
```diff
@@ -0,0 +1,69 @@
+
+test.use({ storageState: 'playwright/.auth/certified-user.json' });
+
+test.describe('Classic challengs - 3 pane desktop layout component', () => {
```
typo

```diff
- challengs 
+ challenge 
```

shabeebk replied:
Fixed.

raditotev commented on e2e/desktop-layout-component.spec.ts:
```diff
@@ -0,0 +1,69 @@
+  });
+});
+
+test.describe('Classic challengs - 2 pane desktop layout component', () => {
```
Same typo `challengs`.

shabeebk replied:
Fixed.

raditotev commented on e2e/desktop-layout-component.spec.ts:
```diff
@@ -0,0 +1,69 @@
+test.use({ storageState: 'playwright/.auth/certified-user.json' });
+
+test.describe('Classic challengs - 3 pane desktop layout component', () => {
+  test.beforeEach(async ({ page }) => {
+    await page.goto(
+      'learn/2022/responsive-web-design/build-a-survey-form-project/build-a-survey-form'
+    );
+  });
```
There is a single test in this `describe` block, `beforeEach` hook is not really needed. It's ok if you decide  to leave it.

shabeebk replied:
Removed the beforeEach hook with [825a9a9](https://github.com/freeCodeCamp/freeCodeCamp/pull/52012/commits/825a9a9c856598bfbf4b9232618dbec93f64967f)

shabeebk committed:
Fix typo

shabeebk committed:
Remove beforeEach from the test

As there is only one test in a description, before each function is not
required. Instead, moved the page.goto inside the test.

shabeebk commented:
Thanks, @raditotev for the comments. Changes made as needed.

Thanks for the approval @naomi-lgbt @Ahmadkashif 🥳

raditotev commented:
Awesome 👍
